src/services/mapCode.ts

```diff
@@ -0,0 +1,329 @@
+import {
+    Block,
+    ClassElement,
+    ClassLikeDeclaration,
+    createSourceFile,
+    emptyArray,
+    FileTextChanges,
+    find,
+    findAncestor,
+    findLast,
+    flatten,
+    forEach,
+    formatting,
+    getTokenAtPosition,
+    isBlock,
+    isClassElement,
+    isClassLike,
+    isForInOrOfStatement,
+    isForStatement,
+    isIfStatement,
+    isInterfaceDeclaration,
+    isLabeledStatement,
+    isNamedDeclaration,
+    isSourceFile,
+    isTypeElement,
+    isWhileStatement,
+    LanguageServiceHost,
+    Mutable,
+    Node,
+    NodeArray,
+    or,
+    ScriptTarget,
+    some,
+    SourceFile,
+    Statement,
+    SyntaxKind,
+    textChanges,
+    TextSpan,
+    TypeElement,
+    UserPreferences,
+} from "./_namespaces/ts.js";
+import { ChangeTracker } from "./textChanges.js";
+
+/** @internal */
+export function mapCode(
+    sourceFile: SourceFile,
+    contents: string[],
+    focusLocations: TextSpan[][] | undefined,
+    host: LanguageServiceHost,
+    formatContext: formatting.FormatContext,
+    preferences: UserPreferences,
+): FileTextChanges[] {
+    try {

```

Comment:
Please remove this try catch now that there aren't any exceptions. We don't want to do this as there are other things you need to worry about if catching exception like cancellation token etc and LS shouldn't be in charge of doing that. If exception happens the tsserver normal exception handling should come in play